ocaml/test/test_platformdata.ml

```diff
@@ -91,27 +96,57 @@ module SanityCheck = Generic.Make(struct
 				"tsc_mode", "2";
 				"viridian", "true";
 				"usb", "true";
-			], false),
-			[
+			], false, 0L, 0L, false),
+			Either.Right ([
 				"usb", "true";
 				"usb_tablet", "false";
 				"parallel", "/dev/parport2";
 				"pae", "true";
 				"tsc_mode", "2";
 				"viridian", "true";
-			]);
+			]));
 			(* Check that combination of valid and invalid fields is dealt with
 			 * correctly. *)
 			(([
 				"pae", "true";
 				"parallel", "/dev/parport0";
 				"tsc_mode", "blah";
-			], false),
-			usb_defaults @
+			], false, 0L, 0L, false),
+			Either.Right (usb_defaults @
 			[
 				"pae", "true";
 				"parallel", "/dev/parport0";
-			]);
+			]));
+			(* Check VCPUs configuration - hvm success scenario*)
+			(([
+				"cores-per-socket", "3";
+			], false, 6L, 6L, true),
+			Either.Right (usb_defaults @
+			[
+				"cores-per-socket", "3";
+			]));
+			(* Check VCPUs configuration - pvm success scenario*)
+			(([
+				"cores-per-socket", "3";
+			], false, 0L, 0L, false),
+			Either.Right (usb_defaults @
+			[
+				"cores-per-socket", "3";
+			]));
+			(* Check VCPUs configuration - hvm failure scenario*)
+			(([
+				"cores-per-socket", "3";
+			], false, 6L, 5L, true),
```

Comment:
Since we're testing the correctness of the value in the platform map, it would be good to also something like

VCPUs_max = 6
VCPUs_at_startup = 6
cores-per-socket = 4